Indent rule errors with array of objects

This issue appeared with v1.1.0.

Rule:
```yml
indent: [2, 2, {"SwitchCase": 1, "VariableDeclarator": 2}]
```

Code:
```js
var tests = [
  { host: 'a.com', cert: { subject: { CN: 'a.com' } } },
  { host: 'a.com', cert: { subject: { CN: 'A.COM' } } },
  {
    host: 'a.com',
    cert: { subject: { CN: 'b.com' } },
    error: 'Host: a.com. is not cert\'s CN: b.com'
  },
  { host: 'a.com', cert: { subject: { CN: 'a.com.' } } },
  { host: 'b.a.com', cert: { subject: { CN: '*.a.com' } } },
  {
    host: 'foo.com', cert: {
      subject: { CN: ['foo.com', 'bar.com'] }
    }
  }
];

```

Error:
```
   4:3  error  Expected indentation of 4 characters but found 2  indent
   5:5  error  Expected indentation of 6 characters but found 4  indent
   6:5  error  Expected indentation of 6 characters but found 4  indent
   7:5  error  Expected indentation of 6 characters but found 4  indent
   8:4  error  Expected indentation of 4 characters but found 2  indent
  11:3  error  Expected indentation of 4 characters but found 2  indent
  12:5  error  Expected indentation of 6 characters but found 4  indent
  15:4  error  Expected indentation of 4 characters but found 2  indent
```

Cross-reference: https://github.com/nodejs/io.js/pull/2286

TLDR: when linting an array literal with object literals, the indent rule with `{"VariableDeclarator": 2}` will always produce a lint error, complaining that it expected a different level of indentation.  If the user changes the level of indentation to the linter's "expectation", they find themselves in an endless loop of errors and suggestions.

So here's my test case:  In my directory I have `js_indent_lint.js` and `eslintrc_indent.json`.

the file being linted, `js_indent_lint.js`:

```
var items = [
  {
    foo: 'bar'
  }
];
```

and the eslint rules file `eslintrc_indent.json`:

```
{
  "rules": {
    "indent": [2, 2, {"VariableDeclarator": 2}]
  }
}
```

When linting this file with this rule, I expect to get no lint errors.  Instead, I get this:

```
$ eslint -c eslintrc_indent.json js_indent_lint.js

js_indent_lint.js
  2:3  error  Expected indentation of 4 characters but found 2  indent
  3:5  error  Expected indentation of 6 characters but found 4  indent
  4:4  error  Expected indentation of 4 characters but found 2  indent

✖ 3 problems (3 errors, 0 warnings)
```

[longer testing story follows]

So I might reasonably expect that I can fix the lint errors by changing the indentation as the output suggests, increasing the indentation by two spaces on lines 2-4, to produce this new `js_indent_lint.js`:

```
var items = [
    {
      foo: 'bar'
    }
];
```

Now running eslint gives me this:

```
$ eslint -c eslintrc_indent.json js_indent_lint.js

js_indent_lint.js
  2:5  error  Expected indentation of 2 characters but found 4  indent

✖ 1 problem (1 error, 0 warnings)
```

Which is curious, because that was the original indentation for line 2, but the errors on the other lines are gone, so let's go ahead and make that last change even though it looks weird:

```
var items = [
  {
      foo: 'bar'
    }
];
```

And re-run eslint:

```
$ eslint -c eslintrc_indent.json js_indent_lint.js

js_indent_lint.js
  2:3  error  Expected indentation of 4 characters but found 2  indent

✖ 1 problem (1 error, 0 warnings)
```

So now eslint suggests changing the indentation back to what it was before.  If I do that, eslint will suggest changing the indentation back to what it is now.  In this test case, eslint produces an endless loop of lint errors.

This only happens when the indent rule includes the `{"VariableDeclarator": 2}` option.

--- a/lib/rules/indent.js
+++ b/lib/rules/indent.js
@@ -142,7 +142,10 @@ module.exports = function(context) {
     function checkNodesIndent(nodes, indent, excludeCommas) {
         nodes.forEach(function(node) {
             var nodeIndent = getNodeIndent(node, false, excludeCommas);
-            if (nodeIndent !== indent && isNodeFirstInLine(node)) {
+            if (
+                node.type !== "ArrayExpression" && node.type !== "ObjectExpression" &&
+                nodeIndent !== indent && isNodeFirstInLine(node)
+            ) {
                 report(node, indent, nodeIndent);
             }
         });

--- a/tests/lib/rules/indent.js
+++ b/tests/lib/rules/indent.js
@@ -615,6 +615,15 @@ ruleTester.run("indent", rule, {
                 "  }\n" +
                 "}\n",
             options: [2, { SwitchCase: 1 }]
+        },
+        {
+            code:
+                "var items = [\n" +
+                "    {\n" +
+                "      foo: 'bar'\n" +
+                "    }\n" +
+                "];\n",
+            options: [2, {"VariableDeclarator": 2}]
         }
     ],
     invalid: [

